Assets/BossRoom/Scripts/Shared/Game/Entity/CustomParentingHandler.cs

```diff
@@ -0,0 +1,158 @@
+using UnityEngine;
+using Unity.Netcode;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// This class temporarily allows NetworkObjects to be parented to children of NetworkObjects, since Netcode for
+    /// GameObjects currently expects NetworkObjects to be parented to other NetworkObjects.
+    /// <remarks>
+    /// To enable a NetworkObject to be parented to this NetworkBehaviour, make sure its "Auto Object Parent Sync"
+    /// field is toggled off.
+    /// </remarks>
+    /// </summary>
+    public class CustomParentingHandler : NetworkBehaviour
+    {
+        public bool preserveWorldSpace = true;
+
+        NetworkList<ulong> m_NetworkObjectChildren = new NetworkList<ulong>();
+
+        public bool hasChildren => m_NetworkObjectChildren.Count > 0;
+
+        public override void OnNetworkSpawn()
+        {
+            if (!IsServer)
+            {
+                m_NetworkObjectChildren.OnListChanged += NetworkObjectChildrenOnListChanged;
+            }
+            base.OnNetworkSpawn();
+        }
+
+        public override void OnNetworkDespawn()
+        {
+            m_NetworkObjectChildren.OnListChanged -= NetworkObjectChildrenOnListChanged;
+        }
+
+        void NetworkObjectChildrenOnListChanged(NetworkListEvent<ulong> changeEvent)
+        {
+            switch (changeEvent.Type)
+            {
+                case NetworkListEvent<ulong>.EventType.Add:
+                    {
+                        AddChildNetworkObject(changeEvent.Value);
+                        break;
+                    }
+                case NetworkListEvent<ulong>.EventType.RemoveAt:
+                case NetworkListEvent<ulong>.EventType.Remove:
+                    {
+                        RemoveChildNetworkObject(changeEvent.Value);
+                        break;
+                    }
+                case NetworkListEvent<ulong>.EventType.Clear:
+                    {
+                        RemoveAllChildren();
+                        break;
+                    }
+            }
+        }
+
+        void AddChildNetworkObject(ulong networkObjectId)
+        {
+            if (NetworkManager.SpawnManager.SpawnedObjects.ContainsKey(networkObjectId))
+            {
+                if (!OnTrySetParent(NetworkManager.SpawnManager.SpawnedObjects[networkObjectId]))
+                {
+                    //Error message
```

Comment:
?
empty if?

Reply:
@VALERE91Unity yes, this class will ideally be removed once SDK allows for parenting a NetworkObject to a child of another NetworkObject. Error messages have been added.

Reply:
@VALERE91Unity will it be fixed for this release? if yes, let's not add a workaround we know will disappear very soon.
if not for this release, then we don't have a choice to implement the workaround